pennylane/ops/qubit/hamiltonian.py

```diff
@@ -222,7 +222,12 @@ def __init__(
         # create the operator using each coefficient as a separate parameter;
         # this causes H.data to be a list of tensor scalars,
         # while H.coeffs is the original tensor
+
         super().__init__(*coeffs_flat, wires=self._wires, id=id)
+        try:
+            self._pauli_rep = qml.pauli.pauli_sentence(self)
+        except ValueError:
+            self._pauli_rep = None
```

Comment:
The try..except could have some performance implications, could we instead do it similarly to how `Sum` builds it, i.e. check if all ops have a pauli_rep? And then either do the `qml.pauli.pauli_sentence` function for conversion or even just by hand.

Reply:
Actually good point about the performance. I'm going to defer calculating the pauli rep until it's requested to fix that potential performance issue.

Reply:
ah nice! would that mean that `op.pauli_rep` is None when checked initially?
Also wondering how performance is calculating hamiltonian pauli reps

Reply:
`op._pauli_rep = "unset"` until `pauli_rep` is requested.